### PR TITLE
Fixes #26676 - improve setting initialization

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Followup for #6737, but took bit different aproach for the caching.

We will need to make changes in all plugins with this aproach.
I will file the PRs after some agreement on this one.
Good think is, that if we do not change the plugins, they will only load the settings twice, what is not that bad.

### CODE REVIEW

**Comment by ares** [comment]
[test foreman] [test katello]

**Comment by lzap** [comment]
Tests are green, @ares are you happy with the code? I can test.

**Comment by ares** [comment]
I'm happy with the code :-)

**Review by lzap — APPROVED** [review]
It works fine, I can only see three saves during initialization:

```
********************************************** SAVE
[app|W||] Failed to remove websockets_ssl_key from cache
********************************************** SAVE
[app|W||] Failed to remove websockets_ssl_cert from cache
********************************************** SAVE
[app|W||] Failed to remove websockets_encrypt from cache
```

This is probably caused by readonly settings, anyway this is much better than it was.

Settings do work properly.

File: app/models/setting.rb [hunk 1]
```diff
@@ -269,10 +275,19 @@ def self.cache
 
   # Methods for loading default settings
 
+  def self.default_settings
+    []
+  end
```
Comment (lzap) [inline app/models/setting.rb:280]: One note - this method is new, all plugins which will start using this method must bump the minimum required Foreman version in the `engine.rb` file to 1.24. Can you update all your PRs please?

Comment (ezr-ondrej) [reply app/models/setting.rb:280]: Oh snap.... I don't want to, but I'll :frowning_man:

**Pull request merged**